### PR TITLE
Add client side encryption

### PR DESCRIPTION
**Pull request by chaudyg**
Add the ability to perform client side encryption using [google Tink](https://github.com/google/tink).
This provides the ability for the user to chose the appropriate encryption algorithm, but also simplify complexe operations like key rotation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,4 +44,60 @@ xcaddy run
...
+    }
+  ]
+}
+```
+
+3. Start caddy with the following config
+
+```
+{
+	storage gcs {
```
Comment (filewalkwithme) [inline README.md:84]:
```suggestion
  storage gcs {
```

File: storage_test.go [hunk 2]
```diff
@@ -105,3 +113,42 @@ func TestUnlock(t *testing.T) {
...
+				Name:       "some/object/",
+			},
+		},
+	})
+	kh, err := keyset.NewHandle(aead.AES256GCMKeyTemplate())
+	assert.NoError(t, err)
+	kp, err := aead.New(kh)
+	assert.NoError(t, err)
+
+	s.aead = kp
```
Comment (filewalkwithme) [inline storage_test.go:131]: this assignment seems to be ineffectual, am I missing something?

File: storage_test.go [hunk 3]
```diff
@@ -105,3 +113,42 @@ func TestUnlock(t *testing.T) {
...
+	s.aead = &cleartext{}
```
Comment (chaudyg) [inline storage_test.go:132]:
```suggestion
```

File: storage_test.go (hunk 2 above)
Comment (chaudyg) [reply storage_test.go:131]: Good point!

**Review by filewalkwithme — APPROVED** [review]
LGTM

**Pull request merged**